lib/index.js

```diff
@@ -228,3 +236,35 @@ FacebookPixel.prototype.completedOrder = function(track) {
 function formatRevenue(revenue) {
   return Number(revenue || 0).toFixed(2);
 }
+
+/**
+ * Get Traits Formatted Correctly for FB.
+ * 
+ * https://developers.facebook.com/docs/facebook-pixel/pixel-with-ads/conversion-tracking#advanced_match
+ * 
+ * @api private
+ */
+
+function formatTraits(analytics) {
+  var traits = analytics && analytics.user().traits();
+  if (!traits) return {};
+  var nameArray = traits.name && traits.name.toLowerCase().split(' ') || [];
+  var gender = traits.gender && traits.gender.slice(0,1).toLowerCase();
+  var birthday = traits.birthday && moment(traits.birthday).format('YYYYMMDD');
+  
+  var address = traits.address || {};
+  var city = address.city && address.city.split(' ').join('').toLowerCase();
+  var state = address.state && address.state.toLowerCase();
+  var postalCode = address.postalCode;
+  return reject({
+    em: traits.email,
+    fn: nameArray.shift(),
```

Comment:
This isn't running off of a message though so it doesn't get to use facade.

This code is running on page load so it's checking for the traits object on the window which is why there's no facade here : (